modelon/impact/client/jupyterhub/authorize.py

```diff
@@ -26,25 +35,25 @@ def authorize(uri, interactive, context=None, credential_manager=None, service=N
         ],
         interactive_help_text=help_text,
     )
-    context = sal.JupyterContext(base=context)
+    jupyter_context = sal.JupyterContext(base=context)
     service = service or sal.JupyterHubService()
 
     try:
-        context.token = _get_jupyter_token(credential_manager, interactive)
-        user = service.get_user_data(uri, context)
+        jupyter_context.token = _get_jupyter_token(credential_manager, interactive)
+        user = service.get_user_data(uri, jupyter_context)
     except exceptions.JupyterHubAuthrizationError:
         if interactive:
             logger.warning(
                 "Could not authorize with the provided JupyterHub API token, "
                 "please enter a new token"
             )
-            context.token = credential_manager.get_key_from_prompt()
-            user = service.get_user_data(uri, context)
+            jupyter_context.token = credential_manager.get_key_from_prompt()
+            user = service.get_user_data(uri, jupyter_context)
         else:
             raise
 
-    if interactive:
-        credential_manager.write_key_to_file(context.token)
+    if interactive and jupyter_context.token:
+        credential_manager.write_key_to_file(jupyter_context.token)
```

Comment:
It seems like pyright is not smart enough to get this check, maybe we could create a separate variable like:
```
jupyter_token = jupyter_context.token
```
to help it?